Spigot-Server-Patches/0619-Added-SoundBuilder.patch

```diff
@@ -0,0 +1,310 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Fruxz <cedricspitzer@outlook.de>
+Date: Wed, 30 Dec 2020 02:23:06 +0100
+Subject: [PATCH] Added SoundBuilder
+
+
+diff --git a/src/main/java/io/papermc/paper/sound/PaperSoundBuilder.java b/src/main/java/io/papermc/paper/sound/PaperSoundBuilder.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..cb6c766f12f5db253d0d09c8a973755236041bd7
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/sound/PaperSoundBuilder.java
+@@ -0,0 +1,237 @@
++package io.papermc.paper.sound;
++
++import com.google.common.collect.Lists;
++import net.minecraft.server.BaseBlockPosition;
++import net.minecraft.server.MinecraftKey;
++import net.minecraft.server.PacketPlayOutCustomSoundEffect;
++import net.minecraft.server.Vec3D;
++import org.bukkit.Location;
++import org.bukkit.Sound;
++import org.bukkit.SoundCategory;
++import org.bukkit.World;
++import org.bukkit.craftbukkit.entity.CraftPlayer;
++import org.bukkit.entity.Player;
++import org.bukkit.util.NumberConversions;
++
++import java.util.ArrayList;
++import java.util.Collection;
++import java.util.List;
++
++public class PaperSoundBuilder implements SoundBuilder {
++
++    private Sound sound;
++    private SoundCategory category;
++    private Collection<Player> receivers;
++    private Location location;
++    private float volume = 1F;
++    private float pitch = 1F;
++
++    public PaperSoundBuilder(Sound sound, Location location) {
++        this.sound = sound;
++        this.location = location;
++    }
++
++    public PaperSoundBuilder(Sound sound, Location location, float volume) {
++        this.sound = sound;
++        this.location = location;
++        this.volume = volume;
++    }
++
++    public PaperSoundBuilder(Sound sound, Location location, Float volume, float pitch) {
++        this.sound = sound;
++        this.location = location;
++        this.volume = volume;
++        this.pitch = pitch;
++    }
++
++    public PaperSoundBuilder(Sound sound, Location location, SoundCategory category) {
++        this.sound = sound;
++        this.location = location;
++        this.category = category;
++    }
++
++    public PaperSoundBuilder(Sound sound, Location location, SoundCategory category, float volume) {
++        this.sound = sound;
++        this.location = location;
++        this.category = category;
++        this.volume = volume;
++    }
++
++    public PaperSoundBuilder(Sound sound, Location location, SoundCategory category, float volume, float pitch) {
++        this.sound = sound;
++        this.location = location;
++        this.category = category;
++        this.volume = volume;
++        this.pitch = pitch;
++    }
```

Comment:
```suggestion
+    public PaperSoundBuilder(SoundCategory soundCategory, Location location) {
+        this.soundCategory = soundCategory;
+        this.location = location;
+    }
```

The only thing you throw or threw an IllegalStateException for is SoundCategory and Location so it seems that both are the required one. So add only one constructor and everything else is optional.